### PR TITLE
GEODE-8918: fix replication in multisite systems

### PR DESCRIPTION
**Pull request by mkevo**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by DonalEvans — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java [hunk 1]
```diff
@@ -741,8 +741,8 @@ public boolean isPdxTypesRegion() {
         // This is for all regions except pdx Region
         if (!isPdxTypesRegion) {
           // Make sure we are distributing to only those senders whose id
-          // is available on this region
-          if (allGatewaySenderIds.contains(sender.getId())) {
+          // is available on this region and whose state is running
+          if (allGatewaySenderIds.contains(sender.getId()) && sender.isRunning()) {
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java:745]: Please consider writing a unit test for any code that only has integration/distributed test coverage. The following example is using a basic technique from Working Effectively with Legacy Code.

For example, you could create a new method called:
```
  @VisibleForTesting
  static boolean hasRunningGatewaySender(Set<GatewaySender> senders, GatewaySender sender) {
    return senders.contains(sender) && sender.isRunning();
  }
```
And then you can add unit tests for this new method to `AbstractRegionTest`:
```
import java.util.Collections;
import org.apache.geode.cache.wan.GatewaySender;

  @Test
  public void hasRunningGatewaySender_returnsFalse_ifSendersIsEmpty() {
    GatewaySender sender = mock(GatewaySender.class);

    boolean value = AbstractRegion.hasRunningGatewaySender(Collections.emptySet(), sender);

    assertThat(value).isFalse();
  }
```
You can also write it as a non-static method, but the unit tests will require a bit more setUp.

**Review by kirklund — CHANGES_REQUESTED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java (hunk 1 above)
Comment (mkevo) [reply geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java:745]: Done!

File: geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java [hunk 2]
```diff
@@ -741,8 +741,8 @@ public boolean isPdxTypesRegion() {
...
             allRemoteDSIds.add(sender.getRemoteDSId());
           }
         } else { // this else is for PDX region
```
Comment (nabarunnag) [inline geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java:748]: Do we also need a check if sender is running for PDX region

File: geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java (hunk 1 above)
Comment (nabarunnag) [inline geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java:745]: can we use the new method hasRunningGatewaySender

**Review by nabarunnag — DISMISSED** [review]

**Review by kirklund — APPROVED** [review]
Thank you for adding new unit and dunit tests!

File: geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java (hunk 2 above)
Comment (mkevo) [reply geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java:748]: I tested it and it works without checking if sender is running. All entries will be delivered to all remote sites for pdx regions.

File: geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java (hunk 1 above)
Comment (mkevo) [reply geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java]: Done!

**Comment by mkevo** [comment]
@nabarunnag , can you re-review it please?

**Comment by mkevo** [comment]
@gesterzhou, @boglesby, @nabarunnag can you please review this PR?

**Review by gesterzhou — APPROVED** [review]

**Pull request merged**